src/modules/referral/services/cron-service.ts

```diff
@@ -3,43 +3,64 @@ import { Repository } from "typeorm";
 import { InjectRepository } from "@nestjs/typeorm";
 import { AppConfig } from "../../configuration/configuration.service";
 import { Deposit } from "../../scraper/model/deposit.entity";
-import { EnhancedCron } from "../../../utils";
-import { updateStickyReferralAddresses } from "../services/queries";
+import { EnhancedCron, wait } from "../../../utils";
+import { updateStickyReferralAddressesQuery } from "../services/queries";
+import { StickyReferralAddressesMechanism } from "src/modules/configuration";
 
 @Injectable()
 export class ReferralCronService {
   private logger = new Logger(ReferralCronService.name);
+  private semaphore = false;
 
   constructor(
     @InjectRepository(Deposit) private depositRepository: Repository<Deposit>,
     private appConfig: AppConfig,
   ) {}
 
-  // run cron every 8 minutes
-  @EnhancedCron("0 4-59/8 * * * *")
-  async refreshMaterializedViewCron() {
-    this.logger.log("start refreshMaterializedViewCron()");
-    if (this.appConfig.values.enableReferralsMaterializedViewRefresh) {
+  @EnhancedCron("0 */10 * * * *")
+  async startCrons() {
+    if (this.semaphore === true) {
+      return;
+    }
+    this.semaphore = true;
+
+    await this.updateStickyReferralAddresses();
```

Comment:
Sticky referrals updated by a cron have more sense when the database is rebuilt from scratch and a huge number of events are fetched. This would put a lot of pressure on the `DepositReferralConsumer` which updates sticky referrals on each new deposit.

Reply:
So we relieve the pressure by waiting for a 30s cooldown?

Reply:
If so, then the cooldown length is important and we should consider making the length configurable by env var

Reply:
The cooldown is more of an experiment to see if it produces any effects. Usually, after an intensive SQL query is executed (as the update of sticky referrals), Postgres performs some clean-up operations (delete temp files, perform `VACUUM` operation), but these are meant to run in the background and usually they don't impact the other queries. They are more as a maintenance process

Reply:
It would be nice if there was a call you could make to postgres asking it to cleanup and telling you when done.